### ISSUE TITLE
Investigate QEMU

### ISSUE THREAD

**Issue by mihaip** [mihaip/infinite-mac#72]
Would allow Mac OS X 10-10.5 to run ([guide for native build](https://www.emaculation.com/doku.php/ppc-osx-on-qemu-for-osx)).  https://github.com/atrosinenko/qemujs is a port from a few years ago, https://github.com/garrysmith-forks/browservm.js claims to be another (it's a fork of https://github.com/elliott-wen/browservm.js, which disappeared at some point).

**Comment by nilobject** [comment]
qemu can run 9.2.2 as well, which some applications require

**Comment by that-ben** [comment]
(Note: Most applications that require a newer Mac OS than 9.0.4 actually require 9.1 and above.  The reason why SheepShaver stopped at 9.0.4 is that the MMU in Mac OS become a requirement starting at 9.1 and SheepShaver cannot emulate that.  Mac OS 9.2 did not bring any major change to the table.  It's only speed improvements and bug fixes.)

**Comment by onpaws** [comment]
> the MMU in Mac OS become a requirement starting at 9.1 and SheepShaver cannot emulate that.
TIL, thanks for sharing that tidbit. Do you know why SheepShaver cannot emulate MMUs?
I have the impression MMUs have been emulated by other emulators out there, and a bit curious why this is true of SheepShaver.

**Comment by that-ben** [comment]
SheepShaver came out in Mac OS 8 era (1998) when Mac OS 8.1 had just came and no such thing as MMU nor Mac OS 9 existed yet.  I suspect that Christian Bauer programmed SheepShaver in a way that is too hard to make it work with MMU, but I don't really know.  All I can tell you is that more modern emulators (as of now only QEMU) can emulate Mac OS 9.1 and 9.2 though.

**Comment by upintheairsheep** [comment]
Devos50 has a QEMU fork that can boot iOS 1 emulating the iPod Touch 2G: https://github.com/devos50/qemu

**Comment by that-ben** [comment]
Though, can really old iOS apps even be installed on such an emulator, offline? If so, I wonder if there's an online archive of all iOS apps/games since those were signed by Apple (now expired) and are all taken out of the App Store by now.

**Comment by upintheairsheep** [comment]
iOS 1 has no apps, other than system ones, but for apps go to the internet
archive.

**Comment by that-ben** [comment]
Oh yeah, will you look at that: .ipa files huh? That's fascinating.

**Comment by upintheairsheep** [comment]
> Oh yeah, will you look at that: .ipa files huh? That's fascinating. Do you think a QEMU fork will soon be able to emulate those iOS 3, 4 or 5 apps?

in the near future. iOS 1 is very interesting for some due to being the first ever version of iOS, however some guy on r/Apple_Internal/ has emulated an internal testing firmware for the same device, albeit with a few bugs.

**Comment by that-ben** [comment]
Even though I'm repeating myself, I find this truly fascinating.

**Comment by upintheairsheep** [comment]
Yes, emulation wiki on gametech has lots of misc emulation from car head units to smart tvs to misc cellphones to iOS to iPods that do not run iOS.

**Comment by upintheairsheep** [comment]
> Would allow Mac OS X 10-10.5 to run ([guide for native build](https://www.emaculation.com/doku.php/ppc-osx-on-qemu-for-osx)). https://github.com/atrosinenko/qemujs is a port from a few years ago, https://github.com/garrysmith-forks/browservm.js claims to be another (it's a fork of https://github.com/elliott-wen/browservm.js, which disappeared at some point).

By 2035, we would probably be able to run macOS High Sierra or later on the browser

**Comment by that-ben** [comment]
Yes, I'm following this closely, but QEMU.js is extremely slow.  I have yet to find Windows 7 emulated in a page.  They achieved it a couple years ago with Windows XP, but it's an extremely stripped down, barely working (very slow) build.

**Comment by upintheairsheep** [comment]
> Though, can really old iOS apps even be installed on such an emulator, offline? I mean except for using Cydia. If so, I wonder if there's an online archive of all iOS apps/games since those were signed by Apple (now expired) and were all taken out of the App Store since they're 32-bit.

Do you mean Cycada?

**Comment by upintheairsheep** [comment]
Another QEMU fork: https://github.com/lemonjesus/qemu-ipod-nano

**Comment by upintheairsheep** [comment]
> Devos50 has a QEMU fork that can boot iOS 1 emulating the iPod Touch 2G: https://github.com/devos50/qemu

iOS and iPod OS should be included due to being made by Apple and really easy to implement once you have the forks compiled.

**Comment by UrkiMimi** [comment]
I wouldn't expect iOS emulation to be added in general on Infinite Mac since one iOS is completely different from Mac OS X even though they both share the Darwin kernel, two QEMU.js hasn't received an update in four years.

**Comment by UrkiMimi** [comment]
True, it's just like running Mac OS X 10.7 on my iPad

**Comment by upintheairsheep** [comment]
> QEMU.js is way too slow for current hardware. Perhaps in 10 years.

I guess we’ll have every version of macOS 10.0+ (via Internet Archive URLs) however put a giant warning that it does require an extremely powerful computer to work. If anyone has an M2 or M1 series they should test each version out for how fast they are.

**Comment by that-ben** [comment]
I cannot wait for the M3 iMac to come out a few weeks/months (??) and then, I'll gladly test it, but out of curiosity, do you have any instructions to follow to boot, say Mac OS X 10.2 on QEMU.js like how would one proceed to do that?

**Comment by mihaip** [comment]
GitHub issues are not discussion forums. @upintheairsheep please refrain from commenting unless you have specific technical things to say that would help advance the project.

**Comment by Pierrestro** [comment]
we dont need qemu to do this Dingus PC works

**Comment by upintheairsheep** [comment]
QEMU is needed for x86 Mac OS X

**Comment by mihaip** [comment]
https://github.com/ktock/qemu-wasm is encouraging (https://ktock.github.io/qemu-wasm-demo/ has some demos).
